resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -61,10 +61,8 @@ private[spark] class KubernetesClusterSchedulerBackend(
   private val runningExecutorsToPods = new mutable.HashMap[String, Pod]
   // Indexed by executor pod names and guarded by RUNNING_EXECUTOR_PODS_LOCK.
   private val runningPodsToExecutors = new mutable.HashMap[String, String]
-  // TODO(varun): Get rid of this lock object by my making the underlying map a concurrent hash map.
-  private val EXECUTOR_PODS_BY_IPS_LOCK = new Object
-  // Indexed by executor IP addrs and guarded by EXECUTOR_PODS_BY_IPS_LOCK
-  private val executorPodsByIPs = new mutable.HashMap[String, Pod]
+  private val executorPodsByIPs: concurrent.Map[String, Pod] = new
```

Comment:
The trait `concurrent.Map[K,V]` is the correct interface/trait for the variable type, but it is abstract and can't be instantiated directly, so in this case typing the variable makes sense. Using scala's `concurrent.TrieMap` is another option, although the choice is probably arbitrary.
```scala
val executorPodsByIPs: concurrent.Map[String, Pod] = concurrent.TrieMap.empty[String, Pod]
```

https://www.scala-lang.org/api/2.11.8/index.html#scala.collection.concurrent.TrieMap$